cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -1031,6 +1179,58 @@ mod tests {
         }));
     }
 
+    #[test]
+    fn locking_of_v1_manifest_requires_catalog_client() {
+        let (mut env_view, mut flox, _temp_dir_handle) = empty_core_environment();
+        flox.catalog_client = None;
+
+        fs::write(env_view.manifest_path(), r#"version = 1"#).unwrap();
+
+        let err = env_view
+            .lock(&flox)
+            .expect_err("should fail to lock v1 lockfile with pkgdb");
+
+        assert!(matches!(err, CoreEnvironmentError::CatalogClientMissing));
+
+        let mut mock_client = MockClient::new(None::<&str>).unwrap();
+        mock_client.push_resolve_response(vec![]);
+        flox.catalog_client = Option::Some(mock_client.into());
+
+        env_view
+            .lock(&flox)
+            .expect("lock should succeed with catalog client");
+    }
+
+    #[test]
+    fn upgrade_with_catalog_client_requires_catalog_client() {
+        // flox already has a catalog client
+        let (mut env_view, mut flox, _temp_dir_handle) = empty_core_environment();
+        fs::write(env_view.manifest_path(), r#"version = 1"#).unwrap();
+
+        {
+            let mut mock_client = MockClient::new(None::<&str>).unwrap();
+            mock_client.push_resolve_response(vec![]);
+            flox.catalog_client = Option::Some(mock_client.into());
+            env_view.lock(&flox).unwrap();
+        }
+
```

Comment:
nonblocking: the test passes without this block, and I'm not sure what it's testing
```suggestion
```

Reply:
Also if we need a locked environment, that should probably be a fixture

Reply:
Ah yeah, we dont need to lock it with the catalog indeed, we just look at the manifest.